src/Service.Tests/OpenApiDocumentor/StoredProcedureGeneration.cs

```diff
@@ -0,0 +1,346 @@
+        [DataRow("sp2", new string[] { "age" }, new string[] { "number" }, DisplayName = "Validate that a parameter is resolved separately from an equivalently named output result set column.")]
+        [DataTestMethod]
+        public void ValidateRequestBodyContents(string entityName, string[] expectedParameters, string[] expectedParametersJsonTypes)
+        {
+            Dictionary<OperationType, bool> configuredOperations = ResolveConfiguredOperations(_runtimeEntities[entityName]);
+            foreach (OperationType opType in configuredOperations.Keys)
+            {
+                // Validate that the generated OpenAPI document has keys for all REST methods defined
+                // in the runtime config for the stored procedure entity.
+                bool operationRegisteredInOpenApiDoc = _openApiDocument.Paths["/" + entityName].Operations.ContainsKey(opType);
+                Assert.AreEqual(configuredOperations[opType], operationRegisteredInOpenApiDoc);
+
+                // Get and delete operations will not have request bodies, enforced by OpenAPI.NET
+                if (opType is OperationType.Get || opType is OperationType.Delete)
+                {
+                    continue;
+                }
+
+                OpenApiRequestBody requestBody = GetOperationRequestBody(entityName, opType);
+                OpenApiReference schemaComponentReference = GetRequestBodyReference(requestBody);
+                Assert.AreEqual($"{entityName}{OpenApiDocumentor.SP_REQUEST_SUFFIX}", schemaComponentReference.Id, message: SCHEMA_REF_ID_ERROR);
+                Assert.AreEqual($"{SCHEMA_REF_PREFIX}{entityName}{OpenApiDocumentor.SP_REQUEST_SUFFIX}", schemaComponentReference.ReferenceV3);
+
+                // It is possible to get the schema component from a OpenApiResponse object because OpenAPI.NET functionality
+                // auto-resolves the reference to a concrete OpenApiSchema object. However, to avoid testing OpenAPI.NET functionality,
+                // this test looks directly at the OpenAPI document's "components" property to validate presence
+                // and composition of the generated schema component.
+                Assert.IsTrue(_openApiDocument.Components.Schemas.ContainsKey($"{entityName}{OpenApiDocumentor.SP_REQUEST_SUFFIX}"), message: "Unexpected absence of schema component definition.");
+                Dictionary<string, OpenApiSchema> schemaComponentProperties = new(GetSchemaComponentProperties($"{entityName}{OpenApiDocumentor.SP_REQUEST_SUFFIX}"));
+
+                // Validate that the generated properties do not outnumber the count of expected columns.
+                Assert.AreEqual(expectedParameters.Length, schemaComponentProperties.Count, message: "The number of generated parameters is not expected.");
+
+                // Validate column presence and accurate column JSON type.
```

Comment:
was column here intended to be a parameter?

Reply:
yep, for that section of code. Updating.